paddle/fluid/framework/ddim.h

```diff
@@ -126,6 +126,67 @@ class DDim {
 
   std::string to_str() const;
 
+  DDim reshape(const std::vector<int>& shape) const {
```

Comment:
Since the header file only attempts to implement template and inline methods, perhaps it would be more suited to move the implementations of `reshape` and `transpose` to .cpp file

Reply:
done